lib/HeroBonus.h

```diff
@@ -165,13 +165,12 @@ class DLL_LINKAGE CAddInfo : public std::vector<si32>
 #define BONUS_LIST										\
 	BONUS_NAME(NONE) 									\
 	BONUS_NAME(LEVEL_COUNTER) /* for commander artifacts*/ \
-	BONUS_NAME(MOVEMENT) /*both water/land*/			\
-	BONUS_NAME(LAND_MOVEMENT) \
-	BONUS_NAME(SEA_MOVEMENT) \
+	BONUS_NAME(MOVEMENT) /*Subtype is 1 - land, 0 - sea*/ \
+	BONUS_NAME(MOVEMENT_MODIFIER) /*allows to modify hero movement by percentage, 0 - sea, 1 - land*/\

```

Comment:
@IvanSavenko I tried, but we need to check army always for movement. So, actual movement will be (baseMovement + army_speed_bonus)*(100+MovementModifier)/100. In case of sea - yes, we can. But what we should do in case of land, when speed depends on army?

Reply:
@IvanSavenko will try to convert calculations to another bonus.